helix-core/src/main/java/org/apache/helix/controller/stages/ResourceComputationStage.java

```diff
@@ -98,6 +102,15 @@ public void process(ClusterEvent event) throws Exception {
       }
     }
```

Comment:
It's hard to comment with github review, but with your change, I think the whole logic is cleaner. It would either read IdealState for regular resources or read Task config for task resources.
In this case, could you please consider the following changes,
1. Create a TaskResourceComputationStage, because we the reuse logic has been reduced quite a lot. It is no longer reasonable to put all logic in one class.
2. If another class is too much, we should at least make the workflow clean to fit the following structure:
if (taskCache) {
  // process TF Resource Objects...
} else {
  // process Regular Resource Objects...
}
// Read the current state to backfill any resources that need to be removed.

Reply:
One problem is that the task pipeline still needs the regular resource objects due to targeted jobs. Therefore the task pipeline resource-computation needs everything from resource pipeline resource-computation; it makes sense for task pipeline to be "resource pipeline resource computation + reading configs", which is the current logic.

Reply:
Then let's re-org the logic a little bit. So they are not interleaved.

Reply:
I wouldn't say it's interleaved, it's just how the logic should look. With what you proposed, it looks like:
```
if (taskCache) {
  processIdealStates();
  processWorkflowConfigs();
  processJobConfigs();
} else {
  processIdealStates();
}
backfillCurrentStates();
```
If we take out the common factor, we have:
```
processIdealStates();
if (taskCache) {
  processWorkflowConfigs();
  processJobConfigs();
}
backfillCurrentStates();
```
which is exactly what it is now. I think it's clear enough (especially given that we may remove the if block altogether later).

Reply:
Yeah, that's not too bad.
I was thinking if the RESOURCES attribute is still necessary. But might be. For example, for the targeted jobs. So let's keep it for now.